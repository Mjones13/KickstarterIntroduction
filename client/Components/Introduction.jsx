import NavBar from './navbar.jsx';
import Header from './header.jsx';
import Player from './player.jsx';
import Status from './status.jsx';
import OwnerModal from './ownermodal.jsx';
import BackProjectModal from './backprojectmodal.jsx';
import axios from 'axios';
import React from 'react';
import ReactDOM from 'react-dom';

class Introduction extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      project: {},
      owner: {},
      player: {},
      status: {},
      backButtonClicked: false,
      ownerClicked: false,
    }
    this.getProject = this.getProject.bind(this);
    this.ownerClicked = this.ownerClicked.bind(this);
    this.backButtonClicked = this.backButtonClicked.bind(this);
  }

  componentDidMount() {
    this.getProject(10);
  }

  getProject(projectid) {
    //Set to 1 for now, but project id can be passed in to fetch specific project
    axios.get(`http://127.0.0.1:3000/projects/${projectid}`)
    .then(project => {
      this.setState({
        project: project.data[0],
        owner: project.data[0].owner,
        player: project.data[0].player,
        status: {
          backers: project.data[0].status.backers.toLocaleString(),
          currentRaised: project.data[0].status.currentRaised.toLocaleString(),
          deadline: project.data[0].status.deadline,
          targetGoal: project.data[0].status.targetGoal.toLocaleString(),
        },
      })
    })
  }

  ownerClicked() {
    this.setState({ownerClicked: !this.state.ownerClicked})
  }

  backButtonClicked() {
    this.setState({backButtonClicked: !this.state.backButtonClicked})
  }

  render() {
    return (
      <div onKeyDown={event => {
        if (event.keyCode === 27){
          this.setState({
            ownerClicked: false,
            backButtonClicked: false,
          })
        }
      }}>
        <div id="navigation-bar">
          <NavBar />
        </div>
        <div id="header">
          <Header project={this.state.project} owner={this.state.owner} ownerClicked={this.ownerClicked}/>
        </div>
          {this.state.backButtonClicked ?
          <BackProjectModal backButtonClicked={this.backButtonClicked} project={this.state.project} owner={this.state.owner}/>
          : null}
          {this.state.ownerClicked ? 
          <OwnerModal ownerClicked={this.ownerClicked} owner={this.state.owner} player={this.state.player}/>
          : null}
        <div className="player-and-status-container">
          <Player project={this.state.project} player={this.state.player}/><Status stats={this.state.status} backButtonClicked={this.backButtonClicked}/>
        </div>
      </div>
    )
  }
}

export default Introduction;