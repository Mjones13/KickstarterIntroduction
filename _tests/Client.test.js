import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Introduction from '../client/Components/introduction.jsx';
import NavBar from '../client/Components/navbar.jsx';
import Header from '../client/Components/header.jsx';
import Player from '../client/Components/player.jsx';
import Status from '../client/Components/status.jsx';

Enzyme.configure({adapter: new Adapter()});

describe('<Introduction />', () => {
  describe('render()', () => {
    test('renders the main introduction component', () => {
      const wrapper = shallow(<Introduction />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<NavBar />', () => {
  describe('render()', () => {
    test('renders the component for NavBar', () => {
      const wrapper = shallow(<NavBar />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Header />', () => {
  describe('render()', () => {
    test('renders the component for header', () => {
      const wrapper = shallow(<Header />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Player />', () => {
  describe('render()', () => {
    test('renders the component for player', () => {
      const wrapper = shallow(<Player />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Status />', () => {
  describe('render()', () => {
    test('renders the component for status', () => {
      const wrapper = shallow(<Status />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});