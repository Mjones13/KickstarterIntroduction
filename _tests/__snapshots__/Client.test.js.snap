// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> render() renders the component 1`] = `
<div>
  <div
    id="navigation-bar"
  >
    <NavBar />
  </div>
  <div
    id="header"
  >
    <Header
      owner={Object {}}
      project={Object {}}
    />
  </div>
  <div
    className="player-and-status-container"
  >
    <Player
      player={Object {}}
      project={Object {}}
    />
    <Status
      stats={Object {}}
    />
  </div>
</div>
`;

exports[`<Introduction /> render() renders the main introduction component 1`] = `
<div>
  <div
    id="navigation-bar"
  >
    <NavBar />
  </div>
  <div
    id="header"
  >
    <Header
      owner={Object {}}
      project={Object {}}
    />
  </div>
  <div
    className="player-and-status-container"
  >
    <Player
      player={Object {}}
      project={Object {}}
    />
    <Status
      stats={Object {}}
    />
  </div>
</div>
`;

exports[`<NavBar /> render() renders the component for NavBar 1`] = `
<div
  className="main-navigation-bar"
>
  <div
    id="nav-bar-1"
  >
    <a
      className="black-link"
      href="#"
    >
      Explore
    </a>
    <a
      className="black-link"
      href="#"
    >
      Start a project
    </a>
  </div>
  <div
    id="nav-bar-2"
  >
    <img
      height="18.9375"
      src="https://d3mlfyygrfdi2i.cloudfront.net/kickstarter-logo-color.png"
      width="185"
    />
  </div>
  <div
    id="nav-bar-3"
  >
    <a
      className="black-link"
      href="#"
      onClick={[Function]}
    >
      Search 
      <i
        className="fas fa-search"
      />
    </a>
    <a
      className="black-link"
      href="#"
    >
      Sign in
    </a>
  </div>
</div>
`;
